clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -320,10 +320,11 @@ namespace {
     }
 
     std::string getIsOmitted() const override {
-      if (type == "IdentifierInfo *")
-        return "!get" + getUpperName().str() + "()";
-      if (type == "TypeSourceInfo *")
+      StringRef T = type;
+      if (T == "TypeSourceInfo *")
         return "!get" + getUpperName().str() + "Loc()";
+      if (T.ends_with(" *"))
+        return "!get" + getUpperName().str() + "()";
```

Comment:
This might not be sufficient here, it could end in 'const' /etc, right?

Reply:
I think it's not worse than it was before.
If you object, I could as well just directly compare against `type == "Expr *"`, similarly to how it was done before.
Or I'm not sure how to resolve this comment.
Did you think of dropping suffixes like `const`, `volatile`, etc? If so, why?
Out of curiosity, I put there these asserts `assert(!T.ends_with_insensitive("const"))`, `assert(!T.ends_with_insensitive("volatile"))` and they didn't fire when I ran `check-clang`, thus I think we probably keep the logic simple unless we have a use-case to solve.
WDYT?

Reply:
Ah, you're not wrong about this not being worse... I guess the list of things handled in here are finite, since this is a list from a generated .td file.  

I think handling `Expr *` separately (as a separate condition of the `IdentifierInfo *` line) is probably the 'right' way to go, but I think we should have an 'assert' above the return-false to ensure that anyone who adds a new type makes sure it gets handled here.

Reply:
I see. Makes sense. I'll look into this tomorrow.
Thanks for the review!

Reply:
Indeed, there are more pointer things in the "type" field. Different kinds of Decls and could be others as well.
I agree that the safest is to only enable this for "Expr *".